language/move-binary-format/src/file_format.rs

```diff
@@ -651,20 +673,21 @@ impl AbilitySet {
     /// the same ability, the type arguments need `Store`
     pub fn polymorphic_abilities(
         declared_abilities: Self,
-        type_argument_abilities: impl IntoIterator<Item = Self>,
+        type_arguments: impl IntoIterator<Item = (Self, TypeParamKind)>,
     ) -> Self {
         // Conceptually this is performing the following operation:
         // For any ability 'a' in `declared_abilities`
         // 'a' is in the result only if
-        //   for all 'ti' in `type_argument_abilities`, a.required() is a subset of ti
+        //   for all 'ti' in `type_arguments` s.t. ti.kind is not phantom, a.required() is a subset of ti.constraints
```

Comment:
nit, `.kind` and `.constraints` aren't used here. And `constraints` isn't exactly accurate. 

I'd probably just do `.1` and `.0` or do `for all '(ti, usage)' in 'type_arguments' ...`